APMrover2/mode.cpp

```diff
@@ -7,7 +7,8 @@ Mode::Mode() :
     g2(rover.g2),
     channel_steer(rover.channel_steer),
     channel_throttle(rover.channel_throttle),
-    mission(rover.mission)
+    mission(rover.mission),
+    attitude_control(rover.g2.attitude_control)
```

Comment:
We don't need to pull *everything* into the Mode namespace. 

I mostly pulled everything in to avoid code churn (so avoiding changing "mission" to "rover.mission" all over the place.  In Copter a lot of the things referenced will actually move into the Mode object itself - so you would get churn BOTH times - on the Mode conversion, and later when things are cleaned up!

New objects don't need to be moved into the Mode namespace...

But hey, it's a stylistic thing, so whatever :-)